src/cljs/athens/electron.cljs

```diff
@@ -325,6 +325,48 @@
   ;; open or create a new starter db
 
   ;; Watch filesystem, e.g. in case db is updated via Dropbox sync
+  
+    ;; Update index.transit    
+  (defn write-db-index [file filepath]
+    (.. fs (writeFile filepath file (fn [e] (dispatch [:boot/desktop]))))
+  )
+  
+  ;; Create new index.transit db
+  (defn create-db-index [file filepath]
+    (.. fs (writeFile filepath file (fn [e] (dispatch [:fs/open-dialog]))))
+  ) 
+  
+  (defn open-dialog-index
+    "Allow user to open a Backup file."
+    [filepath]
+    (js/alert "Your index.transit is corrupted, please open up a backup")
+    (let [res  (.showOpenDialogSync dialog (clj->js {:properties ["openFile"]
+                                                     :filters    [{:name "Transit" :extensions ["transit.bkp"]}]}))
+          open-file (first res)]
+      (when (and open-file (.existsSync fs open-file))
+        (let [read-db (.readFileSync fs open-file)
+              db-file      (try  (dt/read-transit-str read-db)(catch  :default e ((js/console.error (js/Error. e))
+                                                                                  (open-dialog-index filepath))))              
+              ] 
+           (if (= (:schema db-file) db/schema) 
+                  ((write-db-index read-db filepath))(open-dialog-index filepath))))
```

Comment:
This code is wrong, does it really work?

Reply:
This code `((write-db-index ...))`, why double parentheses?

Reply:
no need for double fixed